relay/network_graph/graph.py

```diff
@@ -154,6 +154,138 @@ def __init__(self,
+            get_balance(edge_data, dst, node),
+            get_interest_rate(edge_data, dst, node),
+            get_interest_rate(edge_data, node, dst),
+            self.timestamp - get_mtime(edge_data))
+
+        fee = imbalance_fee(self.capacity_imbalance_fee_divisor, pre_balance, self.value + sum_fees)
+
+        if sum_fees + fee > self.max_fees:
+            return None
+
+        # check that we don't exceed the creditline
+        capacity = pre_balance + get_creditline(edge_data, node, dst)
+        if self.value + sum_fees + fee > capacity:
+            return None  # creditline exceeded
+
+        return (sum_fees + fee, num_hops + 1)
+
+
+class ReceiverPaysCostAccumulatorSnapshot(alg.CostAccumulator):
+    """This is the CostAccumulator being used when using our 'receiver pays
+    fees' style of payments"
+
+    This sorts by the fees first, then the number of hops.
+
+    We need to pass a timestamp in the constructor. This is being used to
+    compute a consistent view of balances in the trustlines network.
+    """
+    def __init__(self, *, timestamp, value, capacity_imbalance_fee_divisor, max_hops=None, max_fees=None, ignore=None):
+        if max_hops is None:
+            max_hops = math.inf
+        if max_fees is None:
+            max_fees = math.inf
+
+        self.timestamp = timestamp
+        self.value = value
+        self.capacity_imbalance_fee_divisor = capacity_imbalance_fee_divisor
+        self.max_hops = max_hops
+        self.max_fees = max_fees
+        self.ignore = ignore
+
+    def zero(self):
+        return (0, 0)
+
+    def total_cost_from_start_to_dst(
+        self, cost_from_start_to_node, node, dst, edge_data
+    ):
+        if dst == self.ignore or node == self.ignore:
+            return None
+
+        sum_fees, num_hops = cost_from_start_to_node
+
+        if num_hops + 1 > self.max_hops:
+            return None
+
+        # fee computation has been inlined here, since the comment in
+        # Graph._get_fee suggests it should be as fast as possible. This means
+        # we do have some code duplication, but I couldn't use some of the
+        # methods in Graph anyway, because they don't take a timestamp argument
+        # and rather use the current time.
+        #
+        # We should profile this at some point in time.
+        #
+        # For this case the pathfinding is not done in reverse.
+
+        pre_balance = balance_with_interests(
+            get_balance(edge_data, node, dst),
+            get_interest_rate(edge_data, node, dst),
+            get_interest_rate(edge_data, dst, node),
+            self.timestamp - get_mtime(edge_data))
+
+        fee = imbalance_fee(self.capacity_imbalance_fee_divisor, pre_balance, self.value - sum_fees)
+
+        if sum_fees + fee > self.max_fees:
+            return None
+
+        # check that we don't exceed the creditline
+        capacity = pre_balance + get_creditline(edge_data, node, dst)
```

Comment:
This is the wrong creditline. The creditline used here needs to be the one given by dst to node, so `get_creditline(dst, node)`

Reply:
Maybe add a test to be sure.